extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -215,7 +215,16 @@ export class ResourceTypePickerDialog extends DialogBase {
 			this._toolsLoadingComponent.loading = true;
 			this._dialogObject.okButton.enabled = false;
 			Promise.all(
-				this._tools.map(tool => tool.loadInformation())
+				this._tools.map(tool => tool.loadInformation().catch(error => {
+					const errorMessage = tool!.statusDescription || getErrorMessage(error);
+					if (errorMessage) {
+						this._dialogObject.message = {
+							level: azdata.window.MessageLevel.Error,
+							text: errorMessage
+						};
+					}
+					// If we are here that implies that the '.then' block of the enclosing Promise.all method is not called so the okButton will continue to remain disabled.
```

Comment:
This is not correct. By catching the error you make it so the promise resolves completely. So the .then will be called after all of these are finished - regardless of whether an error was thrown or not.

Reply:
Ah ok. I will add a state variable to track when a failure occurred to ensure that controls to proceed in the dialogue are disabled.